### PR TITLE
Add open_on_date filter to enrollments; always warn about enrollment open on project end date

### PR DESCRIPTION
**Pull request by gigxz**
1. Addressing feedback on project update mutation https://github.com/greenriver/hmis-warehouse/pull/2788#discussion_r1100643201
2. Add `open_on_date` filter to enrollments query

### CODE REVIEW

File: drivers/hmis/app/graphql/mutations/update_project.rb [hunk 1]
```diff
@@ -20,17 +20,21 @@ def resolve(id:, input:, confirmed:)
...
       errors = HmisErrors::Errors.new
 
-      funder_count = project.funders.where(end_date: nil).count
-      inventory_count = project.inventories.where(inventory_end_date: nil).count
-      open_enrollments = Hmis::Hud::Enrollment.open_on_date.in_project_including_wip(project.id, project.project_id)
-
-      errors.add :base, :information, severity: :warning, full_message: "Project has #{open_enrollments.count} open #{'enrollment'.pluralize(open_enrollments.count)}." if open_enrollments.present?
-      errors.add :base, :information, severity: :warning, full_message: "#{funder_count} open #{'funder'.pluralize(funder_count)} will be closed." if funder_count.positive?
-      errors.add :base, :information, severity: :warning, full_message: "#{inventory_count} open inventory #{'record'.pluralize(inventory_count)} will be closed." if inventory_count.positive?
+      # If project end date is changing
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/update_project.rb:25]: I think based on your suggestion @eanders this is something we should always warn about if the end date is changed.
Previously it was only flagged if the project was being "closed for the first time" (end date went from nil->value)

Comment (eanders) [reply drivers/hmis/app/graphql/mutations/update_project.rb:25]: That make sense.  I doubt OperatingEndDates change very often once set, but I could see someone adding the wrong year and then correcting it.  No harm in checking whenever it is changed (and present.)

File: drivers/hmis/app/graphql/mutations/update_project.rb [hunk 2]
```diff
@@ -20,17 +20,21 @@ def resolve(id:, input:, confirmed:)
...
+      if input.operating_end_date.present? && project.operating_end_date != input.operating_end_date
+        open_enrollments = Hmis::Hud::Enrollment.open_on_date(input.operating_end_date).in_project_including_wip(project.id, project.project_id)
+        errors.add :base, :information, severity: :warning, full_message: "Project has #{open_enrollments.count} open #{'enrollment'.pluralize(open_enrollments.count)} on the selected end date." if open_enrollments.present?
+      end
+
+      # If project is being "closed" for the first time
+      if project.operating_end_date.blank? && input.operating_end_date.present?
```
Comment (eanders) [inline drivers/hmis/app/graphql/mutations/update_project.rb:32]: I think we can check these too, the same restrictions should apply if you are changing the operating end date than if you are adding it.

Comment (eanders) [reply drivers/hmis/app/graphql/mutations/update_project.rb:32]: Oh, never mind, this is just checking if they are still open, not if the dates make sense.

**Review by eanders — APPROVED** [review]

**Pull request merged**